### PR TITLE
feat: Add 'argo {cron,tempalte,cluster-template} update' command

### PR DESCRIPTION
**Pull request by simster7**
Closes: https://github.com/argoproj/argo-workflows/issues/5464

### CODE REVIEW

**Review by simster7 — COMMENTED** [review]
**WIP**, still needs the whole array of testing

File: test/e2e/cron/basic-update.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+apiVersion: argoproj.io/v1alpha1
```
Comment (simster7) [inline test/e2e/cron/basic-update.yaml:1]: Not a fan of creating a new "`-update`" file for the necessary update, but it seems to be the most robust way to do the test and matches what the user experience would be. The alternative would be editing the original file before "updating" it

File: cmd/argo/commands/cron/update.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+			Namespace: cronWf.Namespace,
+		})
+		if err != nil {
+			log.Fatalf("Failed to get existing cron workflow %q to update: %v", cronWf.Name, err)
+		}
+		cronWf.ResourceVersion = current.ResourceVersion
+		updated, err := serviceClient.UpdateCronWorkflow(ctx, &cronworkflowpkg.UpdateCronWorkflowRequest{
+			Namespace:    cronWf.Namespace,
+			CronWorkflow: &cronWf,
+		})
```
Comment (simster7) [inline cmd/argo/commands/cron/update.go:81]: We always need to `Get` before we `Update` to get the current Resource Version. Even if we wanted to do `Patch`, we'd still need to `Get` to compute the diff.

`kubectl` also (obviously) `Gets` before you can `Update` with `kubectl edit ...`

File: cmd/argo/commands/clustertemplate/update.go [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+		cwftmpls, err := unmarshalClusterWorkflowTemplates(body, cliOpts.strict)
+		if err != nil {
+			log.Fatalf("Failed to parse cluster workflow template: %v", err)
+		}
+		clusterWorkflowTemplates = append(clusterWorkflowTemplates, cwftmpls...)
+	}
+
+	if len(clusterWorkflowTemplates) == 0 {
+		log.Println("No cluster workflow template found in given files")
+		os.Exit(1)
```
Comment (terrytangyuan) [inline cmd/argo/commands/clustertemplate/update.go:57]: I saw both `os.Exit(1)` and `log.Fatalf` are used in many of the CLI implementations. What's the convention to follow here and when should we each one?

**Comment by simster7** [comment]
Back to draft, lower priority

**Comment by lshmouse** [comment]
Any update. It's very nice feature to have.

**Pull request closed without merging**